### ISSUE TITLE
[enhancement

### ISSUE THREAD

**Issue by rbirkelbach** [rstudio/rstudio#1711]
Hello,

would it be possible to add .editorconfig support for rstudio? See here: http://editorconfig.org/
If possible, a second improvement would be to add an .editorconfig file to each new rstudio project with sane defaults (e.g. use Hadley Wickham's styleguide).
This would make collaborating much easier and nicer.

Thanks!

**Comment by jmcphers** [comment]
We're big fans of `.editorconfig` (see e.g. https://github.com/rstudio/rstudio/blob/master/src/cpp/.editorconfig). :-) No ETA yet but this is on our wishlist too!

**Comment by katrinleinweber** [comment]
Would this mean that `StripTrailingWhitespace`, `UseSpacesForTab` and `NumSpacesForTab` be removed from `.Rproj`? If so, the argument behind that would be to separate style settings from others, which may be more personal for each RStudio users, correct?

**Comment by jmcphers** [comment]
`.editorconfig` is part of the project, just like `.Rproj` -- so it doesn't really make it personal for each user. The goal is to keep the style consistent in the project. `.Rproj` helps with that as long as all the collaborators are using RStudio. `.editorconfig` is similar, but it is more flexible (you can set defaults per directory and per filetype, for instance) and is supported by many editors other than RStudio.

**Comment by katrinleinweber** [comment]
Yes, I know. What I'm wondering is whether those non-personal, project-wide editor settings will be _moved_ from `.Rproj` to `.editorconfig`. Currently, the former is sometimes not commited because it _also_ contains settings which seem more individual. Unfortunately, such concerns then make it harder to keep the non-personal style elements consistent across different contributors.

**Comment by rbirkelbach** [comment]
Maybe instead of embedding editorconfig in RStudio, it would be easier (I don't know, it's just an idea) to create a translator from editorconfig to rstudioprojects and the other way around?
